app/code/BVMyBlog/Blog/Block/Post.php

```diff
@@ -0,0 +1,87 @@
+<?php
+declare(strict_types=1);
+
+namespace BVMyBlog\Blog\Block;
+
+use BVMyBlog\Blog\Model\BlogRepository;
+use Magento\Framework\App\Response\RedirectInterface;
+use Magento\Framework\DataObject;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+
+/**
+ * Returns post data
+ */
+class Post extends Template
+{
+    /**
+     * @var RedirectInterface $redirect
+     */
+    private $redirect;
```

Comment:
Please use UrlInterface. RedirectInterface should be used only in Action classes